### ISSUE TITLE
Propagate proxy settings when building docker images

### ISSUE THREAD

**Issue by boazy** [concourse/docker-image-resource#131]
Concourse supports passing the http_proxy family of environment variables from the workers to the containers running on them: https://github.com/concourse/concourse/issues/319

These environment variables do not propagate, however, when building a docker image with docker-image-resource which leaves a pretty large blind spot.
It's possible to work-around the problem by manually specifying the proxy environment variables as `build_args`, but this breaks the clean separation between worker configuration and the descriptive build pipeline.

It's possible to fix the issue in [assets/out](https://github.com/concourse/docker-image-resource/blob/febdb88330e65c2ffb445511821caa370b40506e/assets/out) by adding code to check for proxy environment variables and add them if necessary:

```bash
  if [ -n "$http_proxy" ]; then
    expanded_build_args="$expanded_build_args --build-arg http_proxy=$http_proxy"
  fi
  if [ -n "$https_proxy" ]; then
    expanded_build_args="$expanded_build_args --build-arg http_proxy=$https_proxy"
  fi
  if [ -n "$no_proxy" ]; then
    expanded_build_args="$expanded_build_args --build-arg no_proxy=$no_proxy"
  fi
```
This can be refactored to reduce repetition, of course. This would automagically add proper proxy support, but may result in a nasty warning during the docker image build:
[Warning] One or more build-args [http_proxy https_proxy] were not consumed

I'm not sure if there's a way to avoid this warning, but this behavior could be made optional.

**Comment by jtarchie** [comment]
It seems [legit](https://docs.docker.com/engine/reference/commandline/build/#set-build-time-variables-build-arg). Would accept a PR for this.

**Comment by vito** [comment]
fixed by #141
